### ISSUE TITLE
Os_tips does not work with client-side generated pages

### ISSUE THREAD

**Issue by balat** [ocsigen/ocsigen-start#417]
The semantics is:

 - Display only one tip at a time. If there are several tips in the same page, first display the first one, and the next one only when the previous is closed
 - If we change page before closing all tips, show the tips of the new page

**Comment by vasilisp** [comment]
@balat I am trying to understand/debug this. I hope you have been busy with other stuff and haven't made significant progress on it already :).

**Comment by vasilisp** [comment]
I suspect the initial waiter doesn't wait enough. If I change it to

```ocaml
let%client waiter =
  ref (let%lwt _ = Eliom_client.lwt_onload () in
       let%lwt () = Lwt_js.sleep 3. in
       Lwt.return ())
```

things seem to work. Without the sleep, we do the `appendChild`, but some conflict with other that happens during `onload` (?) prevents the initial tip from ever appearing.

**Comment by vasilisp** [comment]
If my hypothesis stands, we could consider some kind of `Eliom_client.late_onload` and `Eliom_client.lwt_late_onload`. Or a priority mechanism for the onload handlers.

**Comment by vasilisp** [comment]
I implemented a priority mechanism (https://github.com/ocsigen/eliom/commit/66763b2b126ff03bc2fd5cd361fba8f253fc583f) to make that `lwt_onload` happen after other `onload ` handlers / events. But that doesn't work. We have to show the tip even later. No idea when...

**Comment by vasilisp** [comment]
I have a small patch (https://github.com/ocsigen/eliom/commit/a0bbde8e09859087c7b3096b6b64f0e46a91905b and https://github.com/ocsigen/ocsigen-start/commit/b5232bfe3c30d1fb64ef9e4e57c264ab14ed10cd) that re-initializes the waiter before the service handler is called. I did this to implement the semantics for page change, but it seems to fix tips even for the first time, without any sleep. I don't completely understand why, but we seem to be making progress.

**Comment by vasilisp** [comment]
@balat I pushed a fix to the `demos` branch (#418). It needs ocsigen/eliom#475.
